util.loop.js

```diff
@@ -62,21 +62,42 @@ module.exports = {
 
   defendAndRepair() {
     // Increase walls
-    everyTicks(400, function() {
+    const time = 400;
+    everyTicks(time, function() {
       for(let r in Game.rooms) {
         let room = Game.rooms[r];
 
-        if(!room.memory.maxWallHits) {
+        if(!room.memory.maxWallHits && room.isStronghold()) {
           // set initial wall hits
           // TODO: Test - is this start scenario well adjusted
-          if(room.isStronghold()) {
-            room.memory.maxWallHits = 1000;
-          }
+          room.memory.maxWallHits = 1000;
         }
-        // TODO: review 350000 limit
-        else if(room.memory.maxWallHits < 350000) {
-            room.memory.maxWallHits += 1000;
-            Logger.log('Increased walls in ', room.name,' to:',  room.memory.maxWallHits);
+        else if(room.memory.maxWallHits < 3000000) {
+            var oldLimit = room.memory.maxWallHits;
+            const base = 1000;
+
+            let walls = room.walls().length;
+
+            // We increase the added value by controller level
+            if(room.controller.level == 4) {
+                let value = Math.floor((time / 4) / walls) * base;
```

Comment:
Too much. I changed 4 to 8 tomorrow and will push a more stable version at afternoon.